uspace/app/bdsh/compl.c

```diff
@@ -208,15 +209,24 @@ static errno_t compl_init(wchar_t *text, size_t pos, size_t *cstart, void **stat
 			goto error;
 		}
 		*cstart += rpath_sep + 1 - prefix;
-		free(prefix);
-		prefix = NULL;
 
 		cs->path_list = malloc(sizeof(char *) * 2);
 		if (cs->path_list == NULL) {
 			retval = ENOMEM;
 			goto error;
 		}
-		cs->path_list[0] = dirname;
+
+		if (!is_path(prefix) && cs->is_command) {
+			cs->path_list[0] = malloc(sizeof(char) * PATH_MAX);
```

Comment:
Return value from malloc is not tested here. I think C99 snprintf() reacts gracefully to a NULL str, but I am not sure this is also the case of our snprintf. Better test the value returned from malloc() to be on the safe side.